### PR TITLE
Add elevation and area metric plots

### LINKED ISSUE #50: Plot metrics by elevation and area
Per Jeremy's suggestion in [#48](https://github.com/USGS-R/lake-temperature-lstm-static/pull/48#issuecomment-1243961974): 
> Some other potentially interesting dimensions to view error/bias along might include elevation and lake area.

Let's add plots of RMS error and bias for the variables elevation and area.

### PR DESCRIPTION
**Pull request by AndyMcAliley**
Per Jeremy's suggestion in [#48](https://github.com/USGS-R/lake-temperature-lstm-static/pull/48#issuecomment-1243961974): 
> Some other potentially interesting dimensions to view error/bias along might include elevation and lake area.

This PR adds plots of RMS error and bias for the variables elevation and area. The plots can, optionally, include metrics evaluated against the mean of the training data (e.g., for elevation, take the mean temperature of all lakes in each elevation bin and compute RMS error and bias). Every lake has a different elevation and area, so in order to make meaningful plots, elevation and area need to be binned. 

## How to run the code

- Create all validation plots.
```bash
snakemake -c all -r -s 5_visualize.smk 5_visualize/out/model_prep/initial/cpu_a/plot_all_valid.dummy
```

This code has been run locally with success.

### RMSE by elevation over validation set
rmse-by-elevation-over-valid

### Bias by elevation over validation set
bias-by-elevation-over-valid

### RMSE by area over validation set
rmse-by-area-over-valid

### Bias by area over validation set
bias-by-area-over-valid

## How to review this PR

- Are there any obvious errors?

Hopefully this review isn't too time-intensive.

### Issues that will be addressed in upcoming PRs (so don't worry about them yet)

- Change to a different plotting library - probably `seaborn`
- Plot histograms of observation data density per #49 

Closes #50

### CODE REVIEW

**Review by jdiaz4302 — APPROVED** [review]
Looks good, changes seem to minimally modify previously reviewed code - handling the new column names, variable-specific binning, file paths, and documentation

**Comment by jdiaz4302** [comment]
Something that stands out to me on the plots is that RMSE seems to hover above 2 when viewed across elevation values, never really reaching an RMSE of 4 (maybe at one bin), but when RMSE is viewed across lake area, it's strange to me that RMSE seems to peak higher (almost 5) but not have lower minimums (e.g., falling below 2) to average out to the same partition-wide RMSE. I suppose it could have something to do with how populated/dense the bins are

**Comment by AndyMcAliley** [comment]
> it could have something to do with how populated/dense the bins are

Bingo!
obs-count-by-area-over-valid
